gobblin-service/src/main/java/org/apache/gobblin/service/modules/db/ServiceDatabaseProviderImpl.java

```diff
@@ -63,8 +67,10 @@ private synchronized void ensureDataSource() {
     dataSource.setUsername(configuration.getUserName());
     dataSource.setPassword(configuration.getPassword());
 
-    // MySQL server can timeout a connection so we need to validate connections.
-    dataSource.setValidationQuery(MysqlDataSourceUtils.QUERY_CONNECTION_IS_VALID_AND_NOT_READONLY);
+    // MySQL server can timeout a connection so need to validate connections before use
+    final String validationQuery = MysqlDataSourceUtils.QUERY_CONNECTION_IS_VALID_AND_NOT_READONLY;
+    LOG.info("setting `DataSource` validation query: '" + validationQuery + "'");
+    dataSource.setValidationQuery(validationQuery);
     dataSource.setValidationQueryTimeout(5);
```

Comment:
Seems like it only set the setTestOnCreate() to be true, but if we don't config max lifetime which indicate it's infinite, will there be a problem?

Reply:
I actually check that in our pipeline and see we set it to be infinite, so not sure whether that's the reason we see the failure. But if you think it's not related, I'm fine with that

Reply:
I don't expect infinite lifetime for the troubleshooter/issues store to be the cause of the gobblin cluster downtime